demo/dynamic_embedding/seq_and_dense/seq_and_dense.py.py

```diff
@@ -0,0 +1,753 @@
+  if gpus:
+    max_capacity = 2 * vocab_size
+    # HKV use 128 slots per bucket, the key may lost if same bucket has more than 128 keys
+    # so set the factor larger than max_capacity to avoid this case
+    factor = mpi_size * 0.7
+    config = de.HkvHashTableConfig(
+        init_capacity=math.ceil(vocab_size / factor),
+        max_capacity=math.ceil(max_capacity / factor),
+        max_hbm_for_values=math.ceil(max_capacity * value_size * dim / factor))
+    return de.HkvHashTableCreator(config=config, saver=saver)
+  else:
+    # for CuckooHashTable case the init_capacity passed in by Embedding layer
+    # it handles one node multiple gpu but not multi-nodes case
+    return de.CuckooHashTableCreator(saver=saver)
+
+
+class ChannelEmbeddingLayers(tf.keras.layers.Layer):
+
+  def __init__(self,
+               name='',
+               dense_embedding_size=1,
+               sparse_embedding_size=1,
+               dense_dim=1,
+               embedding_initializer=tf.keras.initializers.Zeros(),
+               mpi_size=1,
+               mpi_rank=0):
+
+    super(ChannelEmbeddingLayers, self).__init__()
+    self.dense_embedding_layer = keras.layers.Embedding(
+        input_dim=dense_dim,
+        output_dim=dense_embedding_size,
+        embeddings_initializer=embedding_initializer,
+        name=name + '_DenseUnifiedEmbeddingLayer',
+        dtype=tf.float32)
+
+    init_capacity = 4096000
+    kv_creator_sparse = get_kv_creator(mpi_size, mpi_rank, init_capacity,
+                                       tf.dtypes.float32.size,
+                                       sparse_embedding_size)
+    self.sparse_embedding_layer = de.keras.layers.HvdAllToAllEmbedding(
+        mpi_size=mpi_size,
+        embedding_size=sparse_embedding_size,
+        key_dtype=tf.int64,
+        value_dtype=tf.float32,
+        initializer=embedding_initializer,
+        name=name + '_SparseUnifiedEmbeddingLayer',
+        init_capacity=init_capacity,
+        kv_creator=kv_creator_sparse,
+        short_file_name=True,
+    )
+
+    self.dnn = tf.keras.layers.Dense(
+        128,
+        activation='relu',
+        kernel_initializer=tf.keras.initializers.RandomNormal(0.0, 0.1),
+        bias_initializer=tf.keras.initializers.RandomNormal(0.0, 0.1))
+
+  def __call__(self, features_info):
+    dense_inputs = []
+    sparse_inputs = []
+    sparse_input_dims = []
+    for fea_name, fea_info in features_info.items():
+      # The features of GPU table and CPU table to be combined and queried are processed separately.
+      if fea_info['vocab'] > 0:
+        dense_inputs.append(fea_info['pretreated_tensor'])
+      else:
+        sparse_inputs.append(fea_info['pretreated_tensor'])
+    assert len(sparse_inputs) == 1, "Only one sparse input is supported."
+    dense_input_tensors_concat, row_length = concat_tensors(dense_inputs)
+    dense_emb_concat = self.dense_embedding_layer(dense_input_tensors_concat)
```

Comment:
Generally speaking, before querying the static embedding, a static_lookup will be performed on the input to map it to the range from 0 to N. But now here is good enough for example.